addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/BlankTotpActiveScanRule.java

```diff
@@ -0,0 +1,124 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractHostPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.authhelper.internal.AuthenticationStep;
+
+public class BlankTotpActiveScanRule extends AbstractHostPlugin
+        implements CommonActiveScanRuleInfo {
+    private static final String MESSAGE_PREFIX = "ascanalpha.blanktotp.";
+    private static final Logger LOGGER = LogManager.getLogger(BlankTotpActiveScanRule.class);
+    private static final Map<String, String> ALERT_TAGS = new HashMap<>();
+
+    @Override
+    public int getId() {
+        return 40048;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.MISC;
+    }
+
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+    }
+
+    @Override
+    public String getReference() {
+        return "N/A";
+    }
+
+    @Override
+    public void scan() {
+        try {
+
+            // Get target URL from request
+            HttpMessage msg = getBaseMsg();
+            TotpScanContext context = TotpScanContextHelper.resolve(msg);
+            if (context == null) {
+                return;
+            }
+
+            List<AuthenticationStep> mutableAuthSteps = new ArrayList<>(context.authSteps);
+
+            AuthenticationStep testStep = new AuthenticationStep();
+            testStep.setType(AuthenticationStep.Type.CUSTOM_FIELD);
+            testStep.setXpath(context.totpStep.getXpath());
+            testStep.setCssSelector(context.totpStep.getCssSelector());
+            testStep.setValue("");
+
+            for (int i = 0; i < mutableAuthSteps.size(); i++) {
+                AuthenticationStep step = mutableAuthSteps.get(i);
+                if (step.getType() == AuthenticationStep.Type.TOTP_FIELD) {
+                    mutableAuthSteps.set(i, testStep); // Replace the TOTP step with the test step
+                    break;
+                }
+            }
+            context.browserAuthMethod.setAuthenticationSteps(mutableAuthSteps);
+            context.browserAuthMethod.authenticate(
+                    context.sessionManagementMethod, context.credentials, context.user);
+            boolean webSessionBlankCode = context.browserAuthMethod.wasAuthTestSucessful();
+
+            if (webSessionBlankCode) {
+                buildAlert(
+                                "Blank Passcode Vulnerability",
```

Comment:
All text shown to the user should be i18n'ed. Use code like `Constant.messages.getString(MESSAGE_PREFIX + "soln");`